### PR TITLE
Make use of generated deepcopy instead of copystructure

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
```
BenchmarkCopyStruct-4       	    5301	    228819 ns/op
BenchmarkDeepCopy-4         	  113806	     11320 ns/op
```
The result shows that the deepcopy generated is 20x faster than copystructure

### CODE REVIEW

File: pilot/pkg/model/push_context.go [hunk 1]
```diff
@@ -1421,7 +1425,11 @@ func (ps *PushContext) initDestinationRules(env *Environment) error {
 	destRules := make([]Config, len(configs))
 
 	for i := range destRules {
-		destRules[i] = configs[i].DeepCopy()
+		var tmpCopy networking.DestinationRule
```
Comment (ramaraochavali) [inline pilot/pkg/model/push_context.go]: Is it possible to change the `DeepCopy` implementation to have this logic?

Comment (hzxuzhonghu) [reply pilot/pkg/model/push_context.go]: Yes we can. But i think it is a little tough, have to get the `ConfigMeta.Type`, `ConfigMeta.Group` and `ConfigMeta.Version`, then we need to do each kind of deep copy separately

Comment (ramaraochavali) [reply pilot/pkg/model/push_context.go]: I meant some thing similar - not tried though - but rough sketch, would it work?

```
func (c Config) DeepCopy() Config {
	var clone Config
	cs := proto.Clone(c.Spec)
	clone.ConfigMeta = c.ConfigMeta
	clone.Spec = cs
	return clone
}
```

Comment (hzxuzhonghu) [reply pilot/pkg/model/push_context.go]: Ah, good to know that the DeepCopyInto is implemented by proto.Clone

**Pull request merged**